[Detection Engine] Instrumenting FTR for flaky test investigation

rylnd opened the pull request:
## Summary

🚧 this branch is not intended to be merged, but instead used to exercise the flaky test runner for investigative purposes. 

Previously, https://github.com/elastic/kibana/pull/223043 had this role, but that branch has gathered a lot of noise and we've changed approaches/meandered a bit, there. Instead of rewriting history there, I'm starting a new branch here.

rylnd committed:
Testing whether es_archiver can cause duplicate documents

This integration test is executed 100 times, with the single-document
archive being loaded and unloaded before/after each test. If this test
fails, it will eliminate the detection engine as a culprit.

dgieselaar committed:
[FTR] Instrument FTR with APM

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/yarn_deduplicate'

dgieselaar committed:
Fix project references

dgieselaar committed:
Merge branch 'instrument-ftr' of github.com:dgieselaar/kibana into instrument-ftr

dgieselaar committed:
Consolidate apm configs

dgieselaar committed:
Remove APM overrides

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#8779](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8779)
[❌] x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/general_logic/basic_license_essentials_tier/configs/ess.config.ts: 0/20 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228850%2Fhead)

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8780](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8780)
[✅] x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/general_logic/basic_license_essentials_tier/configs/ess.config.ts: 20/20 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228850%2Fhead)

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#8784](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8784)
[❌] x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/general_logic/basic_license_essentials_tier/configs/ess.config.ts: 66/100 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228850%2Fhead)

rylnd committed:
Properly log test output

rylnd committed:
Give each test run a unique name

rylnd committed:
Set APM outcome to failure on test failure

This should allow us to more clearly trace what's happening via APM
data.

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#8804](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8804)
[❌] x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/general_logic/basic_license_essentials_tier/configs/ess.config.ts: 5/30 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228850%2Fhead)

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#8806](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8806)
[❌] x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/general_logic/basic_license_essentials_tier/configs/ess.config.ts: 4/20 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228850%2Fhead)

rylnd committed:
Log full _search response from elasticsearch

As there might be useful info in here, and using the logging service is
apparently equivalent to `> /dev/null`

rylnd committed:
Annotate current span with es_archiver _bulk response

Again, this should offer more visibility into what the tests are doing.

pheyos committed:
FTR - fix esArchiver duplicate doc ingestion

pheyos committed:
Handle data streams

pheyos committed:
Exclude time series indices from id generation

dmlemeshko committed:
Merge branch 'main' into fix_es_archiver_duplicate_doc

rylnd committed:
Merge branch 'fix_es_archiver_duplicate_doc' into rylnd/instrument-ftr

rylnd committed:
Force a `version_conflict` in our test instead of duplication

By specifying an ID in our archive doc, we should no longer get the
duplication, but I expect instead to see the index request fail in the
state produced by these tests.

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8959](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8959)
[✅] x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/general_logic/basic_license_essentials_tier/configs/ess.config.ts: 200/200 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F228850%2Fhead)

rylnd commented:
Closing as this work has now been completed with https://github.com/elastic/kibana/pull/229457 for `es_archiver` and https://github.com/elastic/kibana/pull/231392 for our internal `dataGeneratorFactory`.
